SWENT-GRP5/SOLVIT
# Fix Seeker Profile UI Navigation

# Purpose:
- This pull redesigns the Seeker Profile UI to fit the Figma better and uses the Material Theme Color scheme. 
- For the sake of time and simplicity, this PR was also used to fix a simple bug in the Service Requests Overview (deadline called for it)

- **What issue does it solve?** 
- - This PR resolves inconsistencies in the UI and navigation integration with the rest of the app. 
- - The filters did not respond correctly to different screen sizes.
- **What functionality does it add?** 
- - This PR introduces a dialog window when clicking the log-out icon in the top right corner, to confirm log-out and prevent misclicks. It also introduces a go-back button in the top bar, which navigates to the previous screen.

### Related Issues:
- #247 

# Changes Made:
- Introduced a Scaffold layout using as top bar a similar top bar to other screens
- Go back button
- Log-out button 
- BottomNavigation Menu
- Use of Material Theme color scheme for the colors of the screen
- Edited a few tests with a strong tendency to fail on the CI
- Changed the name of the Service filter to "Services" to make it consistent with other places in the app
- Set proportions of the two boxes between the icon and text and fix these components so they are always on the same line and of the same size

# How to Test:
1. Run the app on a phone or emulator
2. Navigate through all screens.
3. Check that the UI is clear and has enough contrast in both Light and Dark conditions
4. Click on the goBack button
5. Navigate back to the Profile Screen and click on the log-out icon to log-out.

i. Run the app on a phone or emulator
ii. Navigate to RequestsOverviewScreen
iii. Zoom in/Zoom out
iv. Check that the filters respond correctly

<details>
<summary>Visual Helpers</summary>

#### SeekerProfileScreen

##### Light Theme
![Screenshot_20241114_120836](https://github.com/user-attachments/assets/a97ae558-a1ab-435d-b8e5-d8d5ac21fa96)


##### Dark Theme
![Screenshot_20241114_120937](https://github.com/user-attachments/assets/0d04665a-f75f-423c-8b76-865f16516b16)

#### RequestsOverviewScreen

##### Default
![Screenshot_20241115_004245](https://github.com/user-attachments/assets/f01ff55b-add0-4fd6-b7e3-ffe52fb4eb10)

##### Zoom In
![Screenshot_20241115_004312](https://github.com/user-attachments/assets/f7ecf183-5c3d-4fd5-bd00-10fea5c6d30b)


</details>

---

# Checklist:
